sotodlib/io/bookbinder.py

```diff
@@ -316,35 +396,36 @@ def add_acu_summary_info(self, frame, t0, t1):
         t1 : float
             Stop time of frame (unix time), inclusive
         """
-        bl = self.blocks.get('ACU_summary_output')
-        if bl.data:
-            az_mode = bl.data["Azimuth_mode"]
-            acu_summary_times = bl.times
-            m = (t0 <= acu_summary_times) & (acu_summary_times <= t1)
+        az_mode = self.hkdata.az_mode
+        az = self.hkdata.az
+        el = self.hkdata.el
+        if az_mode is not None:
+            m = (t0 <= az_mode.times) & (az_mode.times <= t1)
             if not np.any(m):
                 frame['azimuth_mode'] = 'None'
-            elif 'ProgramTrack' in az_mode[m]:
-                frame['azimuth_mode'] = 'ProgramTrack'  # Scanning
+            elif 'ProgramTrack' in az_mode.data[m]:
+                frame['azimuth_mode'] = 'ProgramTrack' # Scanning
             else:
-                frame['azimuth_mode'] = 'Preset'  # "Slewing"
-
-        bl = self.blocks['ACU_broadcast']
-        if bl.data:
-            az = bl.data["Corrected_Azimuth"]
-            el = bl.data["Corrected_Elevation"]
-            m = (t0 <= bl.times) & (bl.times <= t1)
-            dt = np.diff(bl.times[m]).mean()
-            az_vel = np.diff(az[m]) / dt
-            el_vel = np.diff(el[m]) / dt
-
-            frame['azimuth_velocity_mean'] = np.mean(az_vel)
-            frame['azimuth_velocity_stdev'] = np.std(az_vel)
-            frame['elevation_velocity_mean'] = np.mean(el_vel)
-            frame['elevation_velocity_stdev'] = np.std(el_vel)
+                frame['azimuth_mode'] = 'Preset'  # Slewing
+
+        if az is not None:
+            m = (t0 <= az.times) & (az.times <= t1)
+            if np.any(m):
+                dt = np.diff(az.times[m]).mean()
+                az_vel = np.diff(az.data[m]) / dt
+                frame['azimuth_velocity_mean'] = np.mean(az_vel)
+                frame['azimuth_velocity_stdev'] = np.std(az_vel)
+
+        if el is not None:
+            m = (t0 <= el.times) & (el.times <= t1)
+            if np.any(m):
+                el_vel = np.diff(el.data[m]) / dt
```

Comment:
This relies on dt computed in the "az is not None" block.  Granted it would be difficult for only one of az and el to be None.  But that is not what the block structure suggests ... it implies they are independent.